# Crow's Feet Analysis


## Introduction
### Background on Crow's Feet Wrinkles
Crow's feet, often referred to as laugh lines, are small wrinkles that emerge at the outer corners of the eyes. They are among the first signs of aging to appear on the face, primarily resulting from the loss of collagen and elasticity in the skin. Various factors contribute to the development and deepening of crow's feet, including but not limited to, repeated facial expressions (like smiling or squinting), sun exposure, smoking, and dehydration. These wrinkles can range from fine lines to more pronounced creases, depending on numerous individual factors, including genetics and lifestyle choices.

### Objective of the Study
The primary objective of this study is to assess the reduction in crow's feet severity over a 28-day period across a cohort of subjects and to explore potential correlations between the extent of reduction and demographic factors such as sex and race. The study will also aim to identify any potential differences in the reduction of crow's feet severity across different demographic groups.

## Methodology
### Data Description
The data is divided into two parts:
1. **Crow's Feet Depth Data:** Measurements of crow's feet depth for 32 subjects at three different times (Day 0, Day 14, Day 28).
    - Subject Number:
      - Serves as a unique identifier for each participant in the study.
    - Day 0, Day 14, Day 28 (Severity Scores):
      - Represents the crow's feet severity score for each subject at three timepoints.
        - Day 0: Baseline measurement before any treatment or intervention.
        - Day 14: Intermediate measurement to track progress midway through the study period.
        - Day 28: Final measurement to assess the overall change in severity after 28 days.

1. **Demographic Data:** Information on the sex and race of the subjects.
    - Sex:
      - Indicates the biological sex of each participant.
    - Race: 
      - Categorizes participants into racial groups such as Asian, Caucasian, Black, and Bi-racial.

### Description of the Grading Scale for Crow's Feet Severity

The severity of crow's feet was quantified using a numerical scale at each timepoint. This scale includes:

**0 (No Wrinkles):** No visible signs of crow's feet, indicating no wrinkle development.

**1-3 (Mild Wrinkles):** Slight evidence of wrinkle formation that may be more apparent when smiling or engaging the muscles around the eyes.

**4-7 (Moderate Wrinkles):** Clearly visible wrinkles that are apparent even without facial muscle engagement.

**8-9 (Severe Wrinkles):** Deep and extensive wrinkle formation, indicating significant aging or damage to the skin around the eyes.

### Procedure for Measuring Crow's Feet at Each Timepoint

Participants' crow's feet were evaluated at baseline (Day 0), at the midpoint (Day 14), and at the end of the observation period (Day 28). The procedure for these assessments likely involved:

**Initial Baseline Measurement (Day 0):** Participants’ crow's feet were assessed and recorded to establish a baseline for comparison against future measurements. This initial assessment was crucial for observing the natural state of the crow's feet without any treatment effects.

**Midpoint Assessment (Day 14):** A follow-up assessment was conducted two weeks into the study to observe any early signs of change in the severity of the crow's feet. This would help in understanding the initial response to the treatment or intervention being assessed.

**Final Assessment (Day 28):** The last assessment was performed after 28 days to determine the final state of the crow's feet after the treatment period. This measurement was critical for assessing the efficacy of the intervention over a short-term period.

# Import necessary libraries

In [49]:
from openpyxl import load_workbook
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import wilcoxon
import matplotlib.pyplot as plt

## Loading data and initial data exploration

In [5]:
file_path = 'PCR Test for Stats.xlsx'

# Load the workbook
wb = load_workbook(filename=file_path)

# Get all the sheet names
sheets = wb.sheetnames

# Print the number of sheets and sheet names
print(f"Number of sheets: {len(sheets)}")
print(sheets)

Number of sheets: 2
['Crows feet wrinkles', 'Demographics']


We can see that there are 2 sheets in the file. We will load the data from both sheets and explore the data to understand its structure and contents.

In [6]:
# Loading crows feet wrinkles data from sheet1
wrinkles_data = pd.read_excel(file_path, sheet_name='Crows feet wrinkles', skiprows=1, usecols="B:E")

# Loading demographics data from sheet2
demographics_data = pd.read_excel(file_path, sheet_name='Demographics')

In [7]:
wrinkles_data.head()

Subject Number  Day 0  Day 14  Day 28
0               1      7       5       5
1               2      6       6       5
2               3      6       6       5
3               4      7       5       5
4               5      7       6       6

In [8]:
demographics_data.head()

Subject Number     Sex       Race
0               1    Male      Asian
1               2    Male  Caucasian
2               3    Male  Caucasian
3               4  Female  Caucasian
4               5  Female      Asian

In [9]:
wrinkles_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   Subject Number  32 non-null     int64
 1   Day 0           32 non-null     int64
 2   Day 14          32 non-null     int64
 3   Day 28          32 non-null     int64
dtypes: int64(4)
memory usage: 1.1 KB


Wrinkles data comprises of 32 rows and 4 columns. The columns are: `Subject Number`, `Day 0`, `Day 14`, and `Day 28`. 

The `Subject Number` column contains unique identifiers for each participant, while the `Day 0`, `Day 14`, and `Day 28` columns contain crow's feet measurements at respective timepoints.

In [10]:
demographics_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Subject Number  32 non-null     int64 
 1   Sex             32 non-null     object
 2   Race            32 non-null     object
dtypes: int64(1), object(2)
memory usage: 900.0+ bytes


Demographic data comprises of 32 rows and 3 columns. The columns are: `Subject Number`, `Sex`, and `Race`.

## Statistical Analysis
### Descriptive Statistics

In [11]:
wrinkles_data.describe()

Subject Number      Day 0     Day 14     Day 28
count       32.000000  32.000000  32.000000  32.000000
mean        16.500000   6.031250   5.625000   5.000000
std          9.380832   0.822442   0.609071   0.672022
min          1.000000   5.000000   5.000000   4.000000
25%          8.750000   5.000000   5.000000   5.000000
50%         16.500000   6.000000   6.000000   5.000000
75%         24.250000   7.000000   6.000000   5.000000
max         32.000000   8.000000   7.000000   6.000000

- There are no missing values in the wrinkles dataset
- Data was collected from 32 subjects
- The mean value of the variable decreases progressively over the study period, from 6.03125 at Day 0 to 5.625 at Day 14, and down to 5.0 by Day 28. This indicates a gradual decline in the variable being measured.
- The standard deviation decreases from Day 0 to Day 14, suggesting that the data points are becoming more tightly clustered around the mean. However, the slight increase in standard deviation from Day 14 to Day 28 indicates a slight increase in variability among the measurements at the final time point.
- Both minimum values and maximum values decrease over time as well, indicating a general trend of decreasing crow's feet severity across the study period.
- Q1 stayed the same over the study period, indicating that the bottom quarter of subjects maintain. While Q1 and Q3 decreased, this indicates that the middle and the higher quartiles of the data is shifting towards lower values over time.


In [12]:
demographics_data.describe()

Subject Number
count       32.000000
mean        16.500000
std          9.380832
min          1.000000
25%          8.750000
50%         16.500000
75%         24.250000
max         32.000000

This does not provide any information about the distribution of the data


## Sex Distribution

In [19]:
sex_distribution = demographics_data['Sex'].value_counts()
sex_distribution

Sex
Male      17
Female    15
Name: count, dtype: int64

In [20]:
sex_percentage = demographics_data['Sex'].value_counts(normalize=True)*100
sex_percentage

Sex
Male      53.125
Female    46.875
Name: proportion, dtype: float64

### Sex Breakdown
**Males:** There are 17 males in your sample, making up 53.125% of the total population.

**Females:** There are 15 females, constituting 46.875% of the sample.

# Race Distribution

In [21]:
race_distribution = demographics_data['Race'].value_counts()
race_distribution

Race
Asian        12
Caucasian    12
Black         4
Bi-racial     4
Name: count, dtype: int64

In [22]:
race_percentage = demographics_data['Race'].value_counts(normalize=True) * 100
race_percentage

Race
Asian        37.5
Caucasian    37.5
Black        12.5
Bi-racial    12.5
Name: proportion, dtype: float64

# Race Breakdown:

**Asian:** There are 12 individuals identified as Asian, making up 37.5% of the total population.

**Caucasian:** Similarly, 12 individuals are identified as Caucasian, also representing 37.5% of the sample.

**Black:** 4 individuals are identified as Black, constituting 12.5% of the population.

**Bi-racial:** There are also 4 individuals identified as Bi-racial, which makes up another 12.5% of the sample.

## Merging datasets

In [24]:

# Combining both datasets on 'Subject Number'
crows_feet_data = pd.merge(wrinkles_data, demographics_data, on='Subject Number')


In [25]:
crows_feet_data.head()

Subject Number  Day 0  Day 14  Day 28     Sex       Race
0               1      7       5       5    Male      Asian
1               2      6       6       5    Male  Caucasian
2               3      6       6       5    Male  Caucasian
3               4      7       5       5  Female  Caucasian
4               5      7       6       6  Female      Asian

### Number of cases per timepoint

In [27]:
cases_per_timepoint = crows_feet_data[['Day 0', 'Day 14', 'Day 28']].apply(lambda x: x.notnull().sum())
cases_per_timepoint

Day 0     32
Day 14    32
Day 28    32
dtype: int64

### Means and Standard Deviations for each timepoint

In [28]:
means = crows_feet_data[['Day 0', 'Day 14', 'Day 28']].mean()
std_devs = crows_feet_data[['Day 0', 'Day 14', 'Day 28']].std()

In [29]:
means

Day 0     6.03125
Day 14    5.62500
Day 28    5.00000
dtype: float64

In [30]:
std_devs

Day 0     0.822442
Day 14    0.609071
Day 28    0.672022
dtype: float64

### Means and Standard Deviations for each timepoint grouped by race

In [33]:
means_by_race = crows_feet_data.groupby('Race')[['Day 0', 'Day 14', 'Day 28']].mean()
std_devs_by_race = crows_feet_data.groupby('Race')[['Day 0', 'Day 14', 'Day 28']].std()

In [34]:
means_by_race

Day 0    Day 14    Day 28
Race                                   
Asian      6.083333  5.666667  5.250000
Bi-racial  6.250000  5.750000  4.500000
Black      6.000000  5.750000  5.000000
Caucasian  5.916667  5.500000  4.916667

In [35]:
std_devs_by_race

Day 0    Day 14    Day 28
Race                                   
Asian      0.668558  0.651339  0.753778
Bi-racial  1.500000  0.957427  0.577350
Black      0.816497  0.500000  0.816497
Caucasian  0.792961  0.522233  0.514929

### Means and Standard Deviations for each timepoint grouped by Sex

In [36]:
means_by_sex = crows_feet_data.groupby('Sex')[['Day 0', 'Day 14', 'Day 28']].mean()
std_devs_by_sex = crows_feet_data.groupby('Sex')[['Day 0', 'Day 14', 'Day 28']].std()

In [37]:
means_by_sex

Day 0    Day 14    Day 28
Sex                                 
Female  6.066667  5.600000  4.800000
Male    6.000000  5.647059  5.176471

In [38]:
std_devs_by_sex

Day 0    Day 14    Day 28
Sex                                 
Female  1.032796  0.736788  0.774597
Male    0.612372  0.492592  0.528594

## Change over time analysis

In [48]:
# Absolute changes from Day 0 to Day 14 and Day 0 to Day 28
crows_feet_data['Change_0_to_14'] = crows_feet_data['Day 14'] - crows_feet_data['Day 0']
crows_feet_data['Change_0_to_28'] = crows_feet_data['Day 28'] - crows_feet_data['Day 0']

## Percentage Change Analysis
Percentage change in crow's feet values from Day 0 to Day 14, and from Day 0 to Day 28.

In [47]:
crows_feet_data['percentage_change_14'] = ((crows_feet_data['Day 14'] - crows_feet_data['Day 0']) / crows_feet_data['Day 0']) * 100
crows_feet_data['percentage_change_28'] = ((crows_feet_data['Day 28'] - crows_feet_data['Day 0']) / crows_feet_data['Day 0']) * 100


## Wilcoxon signed-rank test

Wilcoxon signed-rank test is a non-parametric test used when the data cannot be assumed to be normally distributed, making it suitable for a wide range of data types, including ordinal data or non-normally distributed interval data.

It is used on paired samples to statistically evaluate the differences between two related groups. The test will be conducted to compare the crow's feet scores at different timepoints (Day 0 vs. Day 14, and Day 0 vs. Day 28) to determine if there is a significant difference in the crow's feet over the study period.

The purpose of performing these tests is to evaluate whether there is a significant reduction (or otherwise change) in the severity of crow's feet over the 14 and 28 days of the study, compared to the baseline measurements. A significant p-value (commonly < 0.05) would suggest that the treatment or intervention being studied has a statistically significant effect on reducing the severity of crow's feet, rejecting the null hypothesis of no median difference.

In [52]:

# Perform Wilcoxon signed-rank test versus Day 0 for Day 14 and Day 28
wilcoxon_results_day_14 = wilcoxon(crows_feet_data['Day 0'], crows_feet_data['Day 14'])
wilcoxon_results_day_28 = wilcoxon(crows_feet_data['Day 0'], crows_feet_data['Day 28'])


In [54]:
print("Wilcoxon Signed-Rank Test Results for Day 0 vs Day 14:")
print(wilcoxon_results_day_14)

print("\nWilcoxon Signed-Rank Test Results for Day 0 vs Day 28:")
print(wilcoxon_results_day_28)

Wilcoxon Signed-Rank Test Results for Day 0 vs Day 14:
WilcoxonResult(statistic=0.0, pvalue=0.0035104707374242826)

Wilcoxon Signed-Rank Test Results for Day 0 vs Day 28:
WilcoxonResult(statistic=0.0, pvalue=1.3624365570821252e-05)


## References 
- https://medium.com/crows-feet/crows-feet-a-wrinkle-is-worth-a-thousand-smiles-bb8385a9601e
- 